aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -402,10 +402,25 @@ void TensorIterator::compute_types(const TensorIteratorConfig& config) {
     if (common_device == kCPU) {
       // Casts to outputs by creating temporaries of the correct dtype (if needed)
       if (config.cast_common_dtype_to_outputs_ && op.is_output && op.current_dtype != common_dtype_) {
+        TORCH_INTERNAL_ASSERT(op.tensor.defined());
```

Comment:
I'm seeing the `continue` on line 372... not seeing how we get here with `op.tensor` undefined, but maybe I'm missing it

Reply:
You can't get here with `op.tensor()` being undefined, that's why it's an assert :) Actually I missed the continue on line 372, but I was fairly confident (for other reasons) that it was impossible, but not 100% so.